holoviews/core/spaces.py

```diff
@@ -594,16 +594,15 @@ def _validate_key(self, key):
 
     def event(self, trigger=True, **kwargs):
         """
-        This method allows any of the available stream parameters to be
-        updated in an event.
+        This method allows any of the available stream parameters
+        (renamed as appropriate) to be updated in an event.
         """
         stream_params = set(util.stream_parameters(self.streams))
         updated_streams = []
         for stream in self.streams:
-            overlap = set(stream.params().keys()) & stream_params & set(kwargs.keys())
-            if overlap:
-                stream.update(**dict({k:kwargs[k] for k in overlap}, trigger=False))
-                updated_streams.append(stream)
+            rkwargs = util.rename_stream_kwargs(stream, kwargs, reverse=True)
+            stream.update(**dict(rkwargs, trigger=False))
```

Comment:
Thanks, that has helped me realize what we need:

* An initial loop over ``set(kwargs.keys()) - set(all_streams)`` to raise an exception for anything in that difference set: these are kwargs that cannot be applicable to any stream. Previously invalid kwargs were silently ignored and I think this is bad behavior unless you have reasons to say otherwise.

* ``util.rename_stream_kwargs(stream, {k:v for k,v in kwargs.items() if k in set(stream.contents.keys())}, reverse=True)`` to filter the kwargs to the applicable ones.

Do you agree that this is the right approach?